db/import.go

```diff
@@ -196,12 +214,26 @@ func (db *Database) migrateMetadata(docid string, body Body, existingDoc *sgbuck
 				return nil, false, getErr
 			}
 
+			// Reload doc expiry from bucket
+			gocbBucket, ok := db.Bucket.(*base.CouchbaseBucketGoCB)
+			if !ok {
+				return nil, false, fmt.Errorf("Metadata migration requires gocb bucket (%T)", db.Bucket)
+			}
+			expiry, getExpiryErr = gocbBucket.GetExpiry(docid)
+			if getExpiryErr != nil {
+				return nil, false, getExpiryErr
+			}
+
 			// If an xattr exists on the doc, someone has already migrated.  Cancel and return for potential import checking.
 			if len(existingDoc.Xattr) > 0 {
 				updatedDoc, unmarshalErr := unmarshalDocumentWithXattr(docid, existingDoc.Body, existingDoc.Xattr, cas, DocUnmarshalAll)
 				if unmarshalErr != nil {
 					return nil, false, unmarshalErr
 				}
+
+				latestExpiry := time.Unix(int64(expiry), 0)
+				updatedDoc.Expiry = &latestExpiry
```

Comment:
Passing the modified expiry back on document doesn't seem like the right approach here.  document.Expiry is the user-facing representation of expiry, unlike BucketDocument.expiry. 

More importantly, I don't see where the returned doc.Expiry gets applied during subsequent document import.  In general it's tricky to pass an expiry back up the stack to the underlying bucket operation in updateDoc, but there's a mechanism for the callback to send a modified expiry up the stack in the middle of a write (used to support setting expiry via the Sync Function).  It doesn't look like that's being leveraged here, though.

It might be cleaner to just have migrateMetadata to return the updated expiry value, and have the caller use that when requiresImport = true.

Reply:
Since this wasn't caught by any existing tests, we might need to revisit what our options are to test.